http/file_server.ts

```diff
@@ -174,20 +175,51 @@ const MEDIA_TYPES: Record<string, string> = {
 
 /** Returns the content-type based on the extension of a path. */
 function contentType(path: string): string | undefined {
-  return MEDIA_TYPES[extname(path)];
+  const ext = path.split(".").pop();
+  return ext ? MEDIA_TYPES[ext] : undefined;
 }
 
-// Generates a SHA-1 hash for the provided string
-async function createEtagHash(message: string) {
-  const byteToHex = (b: number) => b.toString(16).padStart(2, "00");
-  const hashType = "SHA-1"; // Faster, and this isn't a security sensitive cryptographic use case
-
-  // see: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
-  const msgUint8 = encoder.encode(message);
-  const hashBuffer = await crypto.subtle.digest(hashType, msgUint8);
-  const hashArray = Array.from(new Uint8Array(hashBuffer));
-  const hashHex = hashArray.map(byteToHex).join("");
-  return hashHex;
+const hexLookup: string[] = new Array(0xff);
+for (let n = 0; n <= 0xff; ++n) {
+  const hexOctet = n.toString(16).padStart(2, "0");
+  hexLookup[n] = hexOctet;
+}
+
+function hex(buf: ArrayBuffer): string {
+  const u8arr = new Uint8Array(buf);
+  const len = u8arr.length;
+  let hexOctets = "";
+  for (let i = 0; i < len; ++i) hexOctets += hexLookup[u8arr[i]];
+  return hexOctets;
+}
```

Comment:
Use std/encoding/hex and put this optimized encoder in there

Reply:
Didn't know about encoding/hex, it is equivalent in terms of performance. Though we should add a Uint8Array -> hex string function.